sources/Interop/Windows/winrt/wrl/client/ComPtr`1.Manual.cs

```diff
@@ -4,150 +4,401 @@
         }
 
+        /// <summary>
+        /// Releases the current COM object, if any, and replaces the internal pointer with an input raw pointer.
+        /// </summary>
+        /// <param name="other">The input raw pointer to wrap.</param>
+        /// <remarks>This method will release the current raw pointer, if any, but it will not increment the references for <paramref name="other"/>.</remarks>
         public void Attach(T* other)
         {
             if (ptr_ != null)
             {
-                var @ref = ((IUnknown*)(ptr_))->Release();
+                var @ref = ((IUnknown*)ptr_)->Release();
+
                 Debug.Assert((@ref != 0) || (ptr_ != other));
             }
+
             ptr_ = other;
         }
 
+        /// <summary>
+        /// Returns the raw pointer wrapped by the current instance, and resets the current <see cref="ComPtr{T}"/> value.
+        /// </summary>
+        /// <returns>The raw pointer wrapped by the current <see cref="ComPtr{T}"/> value.</returns>
+        /// <remarks>This method will not change the reference count for the COM object in use.</remarks>
+        public T* Detach()
+        {
+            T* ptr = ptr_;
+
+            ptr_ = null;
+
+            return ptr;
+        }
+
+        /// <summary>
+        /// Increments the reference count for the current COM object, if any, and copies its address to a target raw pointer.
+        /// </summary>
+        /// <param name="ptr">The target raw pointer to copy the address of the current COM object to.</param>
+        /// <returns>This method always returns <see cref="S_OK"/>.</returns>
         public readonly int CopyTo(T** ptr)
         {
             InternalAddRef();
+
             *ptr = ptr_;
+
             return S_OK;
         }
 
-        public readonly int CopyTo(Guid* riid, void** ptr)
+        /// <summary>
+        /// Increments the reference count for the current COM object, if any, and copies its address to a target <see cref="ComPtr{T}"/>.
+        /// </summary>
+        /// <param name="p">The target raw pointer to copy the address of the current COM object to.</param>
+        /// <returns>This method always returns <see cref="S_OK"/>.</returns>
+        public readonly int CopyTo(ComPtr<T>* p)
         {
-            return ((IUnknown*)(ptr_))->QueryInterface(riid, ptr);
+            InternalAddRef();
+
+            *p->ReleaseAndGetAddressOf() = ptr_;
+
+            return S_OK;
         }
 
+        /// <summary>
+        /// Increments the reference count for the current COM object, if any, and copies its address to a target <see cref="ComPtr{T}"/>.
+        /// </summary>
+        /// <param name="other">The target reference to copy the address of the current COM object to.</param>
+        /// <returns>This method always returns <see cref="S_OK"/>.</returns>
+        public readonly int CopyTo(ref ComPtr<T> other)
+        {
+            InternalAddRef();
+
+            fixed (ComPtr<T>* p = &other)
```

Comment:
This pin could also be elided with the right helper. The address isn't actually being used, so releasing the value and assigning the field directly is probably better.

Reply:
Removed all extra fixed blocks in e10f572606582cdb1e485423992a813aecd2bba8 🚀